cmd/jaeger/internal/integration/integration.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"context"
+	"log"
+	"os"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/testbed/testbed"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/provider/fileprovider"
+	"go.opentelemetry.io/collector/otelcol"
+
+	"github.com/jaegertracing/jaeger/cmd/jaeger/internal"
+	"github.com/jaegertracing/jaeger/cmd/jaeger/internal/extension/jaegerstorage"
+	"github.com/jaegertracing/jaeger/cmd/jaeger/internal/integration/datareceivers"
+)
+
+type StorageIntegration struct {
+	ConfigFile string
+}
+
+func (s *StorageIntegration) newDataReceiver(t *testing.T, factories otelcol.Factories) testbed.DataReceiver {
+	fmp := fileprovider.New()
+	configProvider, err := otelcol.NewConfigProvider(
+		otelcol.ConfigProviderSettings{
+			ResolverSettings: confmap.ResolverSettings{
+				URIs:      []string{s.ConfigFile},
+				Providers: map[string]confmap.Provider{fmp.Scheme(): fmp},
+			},
+		},
+	)
+	require.NoError(t, err)
+
+	cfg, err := configProvider.Get(context.Background(), factories)
+	require.NoError(t, err)
+
+	config, ok := cfg.Extensions[jaegerstorage.ID].(*jaegerstorage.Config)
+	require.True(t, ok, "no jaeger storage extension found in the config")
+
+	// Hacky way to get the storage extension name.
+	// This way we don't need to modify this,
+	// when a new storage backend is added.
+	name := ""
```

Comment:
Please explain what you are trying to do. Can we not create the component always with the same name for integration tests?

Reply:
From the architecture at issue #5254, I've designed that the data receiver will utilize the artificial jaeger receiver to pull the traces from the database which is through jaeger storage extension. Because of that, we need to host another jaeger storage extension that lives in the data receiver, and the jaeger receiver will point to that extension.

So, since the jaeger storage extension in the data receiver is a duplication from the collector's extension, I unmarshal the extension config to be used for the data receiver to host it. But now the jaeger receiver needs to know the name of that storage to be set at `TraceStorage`, so I need to scrape it through the config whether it is in Memory/Badger/GRPC/Elasticsearch.

But while explaining, I've thought of another easier way to set the jaeger receiver `TraceStorage`. Which is I just need to get it from the exporter. I'm changing the implementation right now and add the explanation to `newDataReceiver` function.